cpp/src/filling/sequence.cu

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/filling.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/iterator.cuh>
+
+namespace cudf {
+namespace experimental {
+
+namespace detail {
+
+namespace {
+
+// This functor only exists here because using a lambda directly in the tabulate() call generates the cryptic
+// __T289 link error.  This seems to be related to lambda usage within functions using SFINAE.
+template<typename T>
+struct tabulator {
+   cudf::numeric_scalar_device_view<T> n_init;
+   cudf::numeric_scalar_device_view<T> n_step;
+
+   T __device__ operator()(cudf::size_type i){
+      return n_init.value() + (i * n_step.value());
```

Comment:
Hm, a side effect of this is that doing `.value` every time is likely incurring a pointer deference for every operation. If you could store the values directly that may be better, but I don't know how to do that without incurring a D2H memcpy to construct the `tabulator` object.

Reply:
Yeah, that's why I suggested making it `const` is in hopes of it more likely being cached.